tensorflow_io/arrow/python/ops/arrow_dataset_ops.py

```diff
@@ -104,6 +106,10 @@ def __init__(self,
         batch_size or 0,
         dtype=dtypes.int64,
         name="batch_size")
+    if batch_mode not in self.batch_modes_supported:
```

Comment:
@terrytangyuan I generally like using strings for these kinds of options as opposed to enums, but if the other datasets in tfio end up doing something else for a similar `batch_mode` param, then we can change this so they are consistent. With this change, it will fail fast so the user will know right away if they entered an invalid string.